Minor edits to README

Rageking8 opened the pull request:
Summary of changes:
- Added backticks to improve readability
- Prefer single backtick for inline code/term rather that triple backticks
- Fixed a couple of errors (e.g. `Vk::Result` (should be lowercase "v"))
- Tweaked some sentences to correct grammer/improve flow
- Attempted to normalize code snippet conventions (e.g. size of indentation, placement of braces, use of spacing, use of qualified identifiers)
- Added links to headers/guides for first references (subsequent references are just enclosing in backticks with no link)
- Added usage of `IMPORTANT` and `NOTE` alerts

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/Vulkan-Hpp?pullRequest=1832) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/Vulkan-Hpp?pullRequest=1832) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/Vulkan-Hpp?pullRequest=1832) it.</sub>

asuessenbach commented on README.md:
```diff
@@ -362,8 +363,8 @@ if (shaderResult1.result != VK_SUCCESS)
 // std::tie support.
-vk::Result result;
-vk::ShaderModule shaderModule2;
+Result result;
```
I think, instead of removing `vk::` here, we should add it in the above snippets.

asuessenbach commented on README.md:
```diff
@@ -335,18 +336,18 @@ ShaderModule shader1;
 Result result = device.createShaderModule(&createInfo, allocator, &shader1);
 if (result.result != VK_SUCCESS)
```
We should use `vk::Result::eSuccess`, instead of `VK_SUCCESS` here and in the snippets below.

asuessenbach reviewed (commented):
That's some great improvement to the readme! Thanks a lot for your contribution.

There are just a couple of minor issues I'd like you to resolve.

Rageking8 commented on README.md:
```diff
@@ -9,12 +9,12 @@ The goal of the Vulkan-Hpp is to provide header only C++ bindings for the Vulkan
 ## Getting Started
 
-Vulkan-Hpp is part of the LunarG Vulkan SDK since version 1.0.24. Just `#include <vulkan/vulkan.hpp>` and you're ready to use the C++ bindings. If you're using a Vulkan version not yet supported by the Vulkan SDK you can find the latest version of the header [here](https://github.com/KhronosGroup/Vulkan-Hpp/blob/master/vulkan/vulkan.hpp).
+Vulkan-Hpp is part of the [LunarG Vulkan SDK](https://www.lunarg.com/vulkan-sdk/) since version 1.0.24. Just `#include <vulkan/vulkan.hpp>` and you're ready to use the C++ bindings. If you're using a Vulkan version not yet supported by the Vulkan SDK, you can find the latest version of the header [here](vulkan/vulkan.hpp).
```
I presume you meant https://github.com/KhronosGroup/Vulkan-Hpp/tree/main/vulkan? Since the link provided just links back to the same page (assuming the user is reading the README in the repo's root).

Rageking8 commented:
Addressed all feedback given and did another light edit pass over the whole README file. It includes more escaping with backticks, some formatting, and fixing a few more errors found ("`IndexType::eUint3`2" (typo) and "`vk::UniqueHandles`" (should not have a trailing "s")).

asuessenbach commented:
Thanks a lot for this nice improvements of the readme!
